database/factories/ClinicFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Database\Factories;
+
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Illuminate\Support\Collection;
+use Lightit\Clinics\Domain\Models\Clinic;
+use Lightit\Doctors\Domain\Models\Doctor;
+
+/**
+ * @extends Factory<Clinic>
+ */
+class ClinicFactory extends Factory
+{
+    protected $model = Clinic::class;
+
+    public function definition(): array
+    {
+        return [
+            'name' => fake()->company() . ' Medical',
+            'address' => fake()->streetAddress() . ', ' . fake()->city()
+        ];
+    }
+
+    public function withDoctors(int $min, int $max): self
+    {
+        return $this->afterCreating(function (Clinic $clinic) use ($min, $max) {
+            $count = rand($min, $max);
+            $doctors = DoctorFactory::new()->createMany($count);
```

Comment:
```suggestion
    public function withDoctors(int $records): self
    {
        return $this->afterCreating(function (Clinic $clinic) use ($records) {
            $doctors = DoctorFactory::new()->createMany($records);
```
Specifying how many you want to be created works better than leaving it to randomness, in my opinion. It gives you better control over how many records are being created, as well as making it easier to perform assertions (where the number of records matters).

Reply:
But randomness occurs after you specify the number of related elements you want. You only choose a minimum and maximum number of relationships that can exist, then you choose the number of elements in the seeder.